xtask/src/themelint.rs

```diff
@@ -69,33 +69,56 @@ impl Rule {
             check_both: true,
         }
     }
-    fn validate(&self, theme: &Theme, messages: &mut Vec<String>) {
-        let mut found_fg = true;
-        let mut found_bg = true;
-        let mut fg_name = "";
-        let mut bg_name = "";
-        if let Some(fg) = &self.fg {
-            fg_name = fg;
+    fn found_impl(theme: &Theme, find: Option<&'static str>) -> bool {
+        if let Some(fg) = &find {
             if theme.get(fg).fg.is_none() && theme.get(fg).add_modifier == Modifier::empty() {
-                found_fg = false;
+                return false;
             }
         }
-        if let Some(bg) = &self.bg {
-            bg_name = bg;
-            if theme.get(bg).bg.is_none() && theme.get(bg).add_modifier == Modifier::empty() {
-                found_bg = false;
-            }
+        return true;
+    }
+    fn found_fg(&self, theme: &Theme) -> bool {
+        return Rule::found_impl(theme, self.fg);
+    }
+    fn found_bg(&self, theme: &Theme) -> bool {
+        return Rule::found_impl(theme, self.bg);
+    }
```

Comment:
Please check what clippy say, it don't need `return` here.

Reply:
did you accidentally review and old revision? this got refactored with https://github.com/helix-editor/helix/commit/464d7c541823cda2cc3016e9f3edfa84c3a5dca2 (because it was a bug as fg and bg also need to be retrieved from the underlying Style).

Latest commit should be ok for review now. I moved `validate` to static scope in order to have a similar style between `Existence` and `Difference` rules. The validations are now `check_difference` and `check_existence`.